letsencrypt/tests/auth_handler_test.py

```diff
@@ -148,6 +152,66 @@ def test_perform_failure(self):
         self.assertRaises(
             errors.AuthorizationError, self.handler.get_authorizations, ["0"])
 
+    @mock.patch("acme.challenges.HTTP01Response.simple_verify")
+    @mock.patch("acme.challenges.TLSSNI01Response.simple_verify")
+    @mock.patch("letsencrypt.auth_handler.logger")
+    def test_simple_verify(self, mock_logger, mock_tlssni01_verify, mock_http01_verify):
```

Comment:
/me is confused.

I agree that this test basically "touches" parts of the execution and ensures that it is ran correctly. That is because ``simple_verify`` itself doesn't alter the response of other functions using it, including ``_solve_challenges`` (I'm not sure if it should). Thus, I can't understand how we can have an implementation where everything is tested in a single run. Issue https://github.com/letsencrypt/letsencrypt/issues/1409 called for the change from stopping the client on ``simple_verify``'s failure to simply logging a warning.

The sample code you've used would be extremely difficult to exist by mere mistake. Conspicuous intent would be required to fool the tests for something like that to exist. In that sense, most tests can be fooled one way or another, and the only way we can stop that is by having detailed reviews before a PR is merged.